### DISCUSSION TITLE
Can't use routine editor

### DISCUSSION THREAD

**Discussion by ajlardner** [SteveG/underthebar#10] [General]
None of my changes to the routine editor will save. it seems the json for the existing routine i am editing is not the expected format for the hevy api. this may need changes but i am investigating more

**Comment by SteveG** [comment]
can confirm that I'm still using the editor to make changes to my own routines so it's not completely broken.

might be something in the type of exercise/routine you are editing that I've not tried myself so I haven't handled properly. for example - I don't think I've modified this app since hevy added rep-ranges so if you're using those it might not work.

happy to have a quick look if you send me a link to your workout on hevy and/or put the json here.

**Reply by ajlardner** [reply]
I realized after coming back to this it's a terrible bug report. It was a long night of messing with your code and the Hevy api to figure things out. I was able to use the routine editor for some things and now I am better understanding it and I think I was wrong about editing not working, I am now able to get edits to save. However, I was still having issues with the new routine functionality. I got it working now, and unfortunately I don't have any of the JSON that I failed with recorded. However, I was wondering if I was missing something as far as error reporting / logging - I can't find that anywhere but it would be helpful in debugging issues like what I was experiencing with the routine editor.

I mentioned this here: https://github.com/SteveG/underthebar/discussions/9

If there is not already functionality to report or record errors or something similar, would you mind if I raised PR for that functionality? I would like to give it a try. Apologies if you already log errors somewhere obvious and I'm just stupid.

In conclusion, you can scrap this as a bug report and chalk it up to me not getting enough sleep and being overzealous and lazy simultaneously. If you could let me know about the error reporting thing though I would appreciate it!

**Comment by SteveG** [comment]
hey, sorry for taking so long to reply. glad you were able to figure out the problem.

I never got a great deal of interest in this app so I have kind of decided that it's not worth the effort to polish it up properly for error logging and stuff. Not disagreeing that it could be tidier.

I need to do something with PR section on the main page because they removed that call from their API, any ideas for that? Probably just generate the equivalent locally.
